plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -878,23 +878,26 @@ func haveSame(a1, a2 []string) bool {
 
 func GeneralPredicates(pod *v1.Pod, meta interface{}, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
 	var predicateFails []algorithm.PredicateFailureReason
-	fit, reasons, err := noncriticalPredicates(pod, meta, nodeInfo)
+	fit, reasons, err := EssentialPredicates(pod, meta, nodeInfo)
 	if err != nil {
 		return false, predicateFails, err
 	}
 	if !fit {
+		// No need check other predicates if the essential predicates failed
 		predicateFails = append(predicateFails, reasons...)
+		return len(predicateFails) == 0, predicateFails, nil
```

Comment:
That'll change the event message of scheduler/kubelet, we should add a release-note for that if it's the solution.

I'm not sure whether we should return immediately when `EssentialPredicates` failed: in predicates, we did not return immediately when any predicate failed although it is enough to skip the node and has a better performance, maybe @wojtek-t / @timothysc can give some input for that.